File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<int, int> map;
        int start = 0;
        int maxCount = 0;

        for(int end = 0; end < s.size(); end++){

            map[s[end]]++;
            while(map[s[end]]>1){
                map[s[start]]--;
                start++;
            }
            maxCount = max(maxCount, end-start+1);
        }

        return maxCount;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        
        start = 0
        helper = {}
        res = 0
        for end in range (len(s)):
            
            helper[s[end]] = helper.get(s[end], 0) + 1

            while (len(helper)  != (end-start+1)):

                helper[s[start]] -= 1
                if (helper[s[start]] <= 0):
                    del helper[s[start]]

                start += 1
            
            res = max(res, end-start+1)
        
        return res